sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleCounter.java

```diff
@@ -30,16 +35,15 @@ private SdkDoubleCounter(InstrumentDescriptor descriptor, WriteableMetricStorage
 
   @Override
   public void add(double increment, Attributes attributes, Context context) {
-    BoundStorageHandle aggregatorHandle = storage.bind(attributes);
-    try {
-      if (increment < 0) {
-        throw new IllegalArgumentException("Counters can only increase");
-      }
-
-      aggregatorHandle.recordDouble(increment, attributes, context);
-    } finally {
-      aggregatorHandle.release();
```

Comment:
This is already taking place in `storage.recordDouble`.

Reply:
Good catch!